packages/ComponentLibrary/src/components/BasicModal/index.tsx

```diff
@@ -28,22 +30,51 @@ const Modal: React.FC<ModalIProps> = ({
   SaveIcon,
   backgroundGradient,
   isFullScreenEnabled = false,
+  open: externalOpen,
 }) => {
-  const [open, setOpen] = useState<boolean>(false);
+  const [internalOpen, setInternalOpen] = useState<boolean>(false);
   const [isFullScreen, setIsFullScreen] = useState<boolean>(false);
   const theme = useTheme();
   const { styles, sx } = useStyles();
 
-  const handleOpen = () => setOpen(true);
+  const isOpen = externalOpen !== undefined ? externalOpen : internalOpen;
+
+  useEffect(() => {
+    if (externalOpen !== undefined && externalOpen !== internalOpen) {
```

Comment:
Si externalOpen siempre está declarado en algún lugar, este código es suficiente. Si existe el riesgo de que no esté definido en absoluto (no haya sido seteada la prop), quizas seria mejor:
```suggestion
    if (typeof externalOpen !== "undefined" && externalOpen !== internalOpen) {
```